src/lava/proc/dense/process.py

```diff
@@ -144,3 +144,83 @@ def __init__(self,
                          log_config=log_config,
                          learning_rule=learning_rule,
                          **kwargs)
+
+
+class DelayDense(Dense):
```

Comment:
I am thinking about the proliferation of all different variants. It might make the end user's life simpler if we check for non-zero delay and switch to this delayed computation. I will leave this decision up to you.

Reply:
We decided to go this route for now, since we did the same with learning. Delays and learning are also not compatible. Figuring out how we compose processes and provide a simple, but powerful user interface is an ongoing discussion.